partiql-value/src/datum.rs

```diff
@@ -0,0 +1,255 @@
+use crate::{Bag, BindingsName, List, Tuple, Value};
+use std::borrow::Cow;
+use std::error::Error;
+
+pub type DatumLowerError = Box<dyn Error>;
+pub type DatumLowerResult<T> = Result<T, DatumLowerError>;
+
+pub trait Datum<D>
+where
+    D: Datum<D>,
+{
+    #[inline]
+    /// Returns true if and only if Value is to be interpreted as `NULL`
+    #[must_use]
+    fn is_null(&self) -> bool {
+        false
+    }
+
+    #[inline]
+    /// Returns true if and only if Value is to be interpreted as `MISSING`
+    #[must_use]
+    fn is_missing(&self) -> bool {
+        false
+    }
+
+    #[inline]
+    /// Returns true if and only if Value is null or missing
+    #[must_use]
+    fn is_absent(&self) -> bool {
+        self.is_null() || self.is_missing()
+    }
+
+    /// Returns true if and only if Value is an integer, real, or decimal
+    #[must_use]
+    fn is_number(&self) -> bool;
```

Comment:
Will some other methods be added for other types? Like `is_bool`, `is_text`, etc?

Reply:
This is just moving some tests out of `Value` directly and into the `Datum` interface. Perhaps those methods will be needed in the future, yes. But, I imagine better matching of types in partiql-types will be preferred.